src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/sort/SortExpression.java

```diff
@@ -15,16 +18,36 @@ public record SortExpression(
     @Nullable float[] vector,
     @Nullable String vectorize) {
 
+  // TODO: either remove the static factories or make this a class, as a record the ctor is public
+
   public static SortExpression sort(String path, boolean ascending) {
     return new SortExpression(path, ascending, null, null);
   }
 
   public static SortExpression vsearch(float[] vector) {
-    return new SortExpression(DocumentConstants.Fields.VECTOR_EMBEDDING_FIELD, false, vector, null);
+    return new SortExpression(VECTOR_EMBEDDING_FIELD, false, vector, null);
   }
 
   public static SortExpression vectorizeSearch(String vectorize) {
-    return new SortExpression(
-        DocumentConstants.Fields.VECTOR_EMBEDDING_TEXT_FIELD, false, null, vectorize);
+    return new SortExpression(VECTOR_EMBEDDING_TEXT_FIELD, false, null, vectorize);
+  }
+
+  /**
+   * Create a sort that is used when sorting a table column by a vector.
+   *
+   * <p>aaron -30-oct, this is a bit of a hack, but it is a quick way to support sorting by a vector
+   * for tables
+   */
+  public static SortExpression tableVectorSort(String path, float[] vector) {
+    return new SortExpression(path, false, vector, null);
+  }
+
+  public boolean isTableVectorSort() {
```

Comment:
what does this mean, column can not be $vector and $vectorize?

Reply:
yes, discussed for now we are saying a vector search on something not $vector is on a table. We need a big re-architecture for the sort deserializer